filterable-file-log/serializer.lua

```diff
@@ -60,11 +64,46 @@ function _M.serialize(ngx, filters)
     service = ngx.ctx.service,
     api = ngx.ctx.api,
     consumer = ngx.ctx.authenticated_consumer,
+    jwt_claims = jwt_claims,
     client_ip = ngx.var.remote_addr,
     started_at = ngx.req.start_time() * 1000
   }
 end
 
+function get_jwt_claims(headers)
+  if not headers['authorization'] then
+    return nil
+  end
+
+  local token = string.match(headers['authorization'], 'Bearer (.+)')
+  if not token then
+    return nil
+  end
+
+  -- The JWT has three .-delimited parts: header, payload, and signature
+  -- The payload has the claims we're interested in
+  local encoded_claims = stringx.split(token, '.')[2]
+  if not encoded_claims then
+    return nil
+  end
+
+  local claims = ngx.decode_base64(encoded_claims)
+  local parsed_claims, err = cjson.decode(claims)
+  if err then
+    return nil
+  end
+
+  -- only log a subset of claims to avoid logging PII/PHI
+  return whitelist_filter(parsed_claims, {
+    'cid',
+    'exp',
+    'iat',
+    'iss',
+    'jti',
+    'scp'
+  })
```

Comment:
So that was actually intentional: leaving out the param defaults it to nil, meaning those entries will be removed from the table. Looking back, I have no idea why I decided that claims should be treated differently than headers.

I'll just revert the change I made to the whitelist function so that filtered claims will show up like `{"uid": "FILTERED"}`.